Update halftone.py with offsets, oversampling, optional ICC profile use

dwringer opened the pull request:
Adds c_, m_, y_, k_offset toggles, oversampling factor, and optional use of ICC profiles from .icc, .icm files in <invokeai>/color_profiles/.

- Resizing is done as uint8 from 0..255, this might not be ideal?
- When oversampling, the image gets converted like: array->image->four images->four arrays; the middle two steps are redundant.

Hopefully I didn't miss anything else, fell asleep with this *I think* in a correct state.

dwringer committed:
Update halftone.py with offsets, oversampling, optional ICC profile use

Adds c_, m_, y_, k_offset toggles, oversampling factor, and optional use of ICC profiles from .icc, .icm files in <invokeai>/color_profiles/.

JPPhoto committed:
Updated halftone.py

Removed color correction because InvokeAI doesn't currently have a good way of dealing with a node's files/resources. Added oversampling into regular grayscale halftoning.

JPPhoto commented:
I changed it to remove color profiles as InvokeAI doesn't have support for nodes' data/resources at the moment and I feel that coding in an absolute path or anything else we do at this point is going to be rendered obsolete.

JPPhoto committed:
Update halftone.py

Do per-channel resizing rather than global. This may allow the garbage collector to free up memory between channel halftonings.
